clang/test/Frontend/sys_macro_defined_type.cpp

```diff
@@ -0,0 +1,22 @@
+// RUN: %clang_cc1 -I%S/Inputs -fsyntax-only -triple x86_64-linux-gnu %s 2>&1 | FileCheck %s
+
+#include <macro_defined_type.h>
+
+// The diagnostic message is hard-coded as 'noderef' so using a system macro doesn't change the behavior
```

Comment:
This seems like it's going to be a confusing user experience because sometimes the user will see `noderef` and other times they may see `_SYS_NODEREF`. Maybe we should make a second pass an update hard-coded names (perhaps as a follow-up)?